nmpolicy/internal/expander/stateExpander.go

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2021 NMPolicy Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *	  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package expander
+
+import (
+	"fmt"
+	"regexp"
+
+	yaml "sigs.k8s.io/yaml"
+
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/capture"
+	"github.com/nmstate/nmpolicy/nmpolicy/types"
+)
+
+type StateExpander struct {
+	capResolver capture.PathCapturer
+}
```

Comment:
```suggestion
type StateExpander struct {
	lexer Lexer
	parser Parser
	resolver Resolver
	capturedState map[string]types.CaptureState
}
```

Reply:
As we discussed offline. I want the logic to be decoupled, I don't want the expander to be aware of the lexer->parser->resolver process. It will just invoke `capturePathResolver.ResolveCapturePath`.